src/code/z_fireobj.c

```diff
@@ -101,33 +95,35 @@ void FireObj_UpdateStateTransitions(PlayState* play, FireObj* fire) {
     Vec3f dist;
 
     FireObj_StepSize(fire);
-    fire->unk26++;
+    fire->timer++;
     if (fire->ignitionDelay > 0) {
         fire->ignitionDelay--;
     } else if (fire->ignitionDelay == 0) {
         fire->ignitionDelay = -1;
-        if ((fire->state == FIRE_STATE_3) || (fire->state == FIRE_STATE_1)) {
-            nextState = FIRE_STATE_0;
+        if ((fire->state == FIRE_STATE_NOT_LIT) || (fire->state == FIRE_STATE_SHRINKING)) {
+            nextState = FIRE_STATE_GROWING;
         } else {
-            nextState = FIRE_STATE_1;
+            nextState = FIRE_STATE_SHRINKING;
         }
         FireObj_SetState(fire, fire->dynamicSizeStep, nextState);
     }
-    if ((fire->flags & 1) && (fire->state != FIRE_STATE_3) &&
+
+    if ((fire->flags & FIRE_FLAG_1) && (fire->state != FIRE_STATE_NOT_LIT) &&
         WaterBox_GetSurface1_2(play, &play->colCtx, fire->position.x, fire->position.z, &waterY, &waterBox) &&
         (waterY - fire->position.y > 6500.0f * fire->yScale)) {
-        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_3);
+        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_NOT_LIT);
     }
-    if ((fire->flags & 2) && (player->heldItemAction == PLAYER_IA_STICK)) {
+    if ((fire->flags & FIRE_FLAG_2) && (player->heldItemAction == PLAYER_IA_STICK)) {
         Math_Vec3f_Diff(&player->meleeWeaponInfo[0].tip, &fire->position, &dist);
         if (Math3D_LengthSquared(&dist) < 400.0f) {
             sp40 = true;
         }
     }
+
     if (sp40) {
-        if (fire->state == FIRE_STATE_3) {
+        if (fire->state == FIRE_STATE_NOT_LIT) {
             if (player->unk_B28 > 0) {
-                FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_0);
+                FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_GROWING);
             }
         } else if (player->unk_B28 == 0) {
             player->unk_B28 = 0xD2;
```

Comment:
`unk_B28` seems to be the deku stick lit/flame timer in this instance, so these can be dec (of course `unk_B28` seems to have other non-deku stick related purposes at a glance, classic player)